packages/typescript/base.js

```diff
@@ -0,0 +1,439 @@
+module.exports = {
+  rules: {
+    // Require that member overloads be consecutive
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/adjacent-overload-signatures.md
+    "@typescript-eslint/adjacent-overload-signatures": "error",
+
+    // Requires using either T[] or Array<T> for arrays
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/array-type.md
+    "@typescript-eslint/array-type": "error",
+
+    // Disallows awaiting a value that is not a Thenable
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/await-thenable.md
+    "@typescript-eslint/await-thenable": "error",
+
+    // Bans @ts-<directive> comments from being used unless @ts-expect-error, which requires description
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-ts-comment.md
+    "@typescript-eslint/ban-ts-comment": [
+      "error",
+      {
+        "ts-expect-error": "allow-with-description",
+        "ts-ignore": "allow-with-description",
+        "ts-nocheck": true,
+        "ts-check": true,
+        minimumDescriptionLength: 10,
+      },
+    ],
+
+    // We're not concerned about ts-lint: comments as they're not used anywhere in our codebase
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-tslint-comment.md
+    "@typescript-eslint/ban-tslint-comment": "off",
+
+    // Bans specific types from being used, e.g.: {}
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/ban-types.md
+    "@typescript-eslint/ban-types": "error",
+
+    // Ensures that literals on classes are exposed as fields to avoid additional overhead of creating a closure
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/class-literal-property-style.md
+    "@typescript-eslint/class-literal-property-style": ["error", "fields"],
+
+    // Enforces the usage of Record<K, V> type instead of { [key: K]: V }
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-indexed-object-style.md
+    "@typescript-eslint/consistent-indexed-object-style": ["error", "record"],
+
+    // Enforces consistent usage of type assertions
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-assertions.md
+    "@typescript-eslint/consistent-type-assertions": [
```

Comment:
Didn't we talk that type assertions should be avoided at all costs?

Reply:
Here's the concept:

All rules can be switched off, therefore:

- If you have the rule turned on, there might be other rules that are redundant because of it, like in this case.
- If you turned off the rules, you should still get the support of best practices like this

Consider this sentence to be the sentiment of this concept:

I'd rather you didn't do type assertions, but if you did, do them consistently.